spec/identity_doc_auth/lexis_nexis/error_generator_spec.rb

```diff
@@ -0,0 +1,233 @@
+      )
+    end
+
+    it 'DocAuthResult is Failed with multiple different alerts' do
+      error_info = build_error_info(
+        'Failed',
+        build_alerts({name: '2D Barcode Read', result: 'Attention'},
+                     {name: 'Visible Pattern', result: 'Failed'}),
+        nil)
+
+      output = described_class.new(config).generate_trueid_errors(error_info, false)
+
+      expect(output.keys).to contain_exactly(:general)
+      expect(output[:general]).to contain_exactly(
+        i18n.t('doc_auth.errors.lexis_nexis.general_error_no_liveness')
+      )
+    end
+
+    it 'DocAuthResult is Failed with multiple id alerts' do
+      error_info = build_error_info(
+        'Failed',
+        build_alerts({name: 'Expiration Date Valid', result: 'Attention'},
+                     {name: 'Full Name Crosscheck', result: 'Failed'}),
+        nil)
+
+      output = described_class.new(config).generate_trueid_errors(error_info, false)
+
+      expect(output.keys).to contain_exactly(:id)
+      expect(output[:id]).to contain_exactly(
+        i18n.t('doc_auth.errors.lexis_nexis.general_error_no_liveness')
+      )
+    end
+
+    it 'DocAuthResult is Failed with multiple back alerts' do
+      error_info = build_error_info(
+        'Failed',
+        build_alerts({name: '2D Barcode Read', result: 'Attention'},
+                     {name: '2D Barcode Content', result: 'Failed'}),
+        nil)
+
+      output = described_class.new(config).generate_trueid_errors(error_info, false)
+
+      expect(output.keys).to contain_exactly(:back)
+      expect(output[:back]).to contain_exactly(
+        i18n.t('doc_auth.errors.lexis_nexis.multiple_back_id_failures')
+      )
+    end
+
+    it 'DocAuthResult is Failed with an unknown alert' do
+      error_info = build_error_info(
+        'Failed',
+        build_alerts({name: 'Not a known alert', result: 'Failed'}),
+        nil)
+
+      expect(exception_notifier).to receive(:call).
+        with(anything, hash_including(:response_info)).twice
+
+      output = described_class.new(config).generate_trueid_errors(error_info, false)
+
+      expect(output.keys).to contain_exactly(:general)
+      expect(output[:general]).to contain_exactly(
+        i18n.t('doc_auth.errors.lexis_nexis.general_error_no_liveness')
+      )
+    end
+
+    it 'DocAuthResult is Failed with multiple alerts including an unknown' do
+      error_info = build_error_info(
+        'Failed',
+        build_alerts({name: 'Not a known alert', result: 'Failed'},
+                     {name: 'Birth Date Crosscheck', result: 'Failed'}),
+        nil)
+
+      expect(exception_notifier).to receive(:call).
+        with(anything, hash_including(:response_info)).once
+
+      output = described_class.new(config).generate_trueid_errors(error_info, false)
+
+      expect(output.keys).to contain_exactly(:id)
+      expect(output[:id]).to contain_exactly(
+        i18n.t('doc_auth.errors.lexis_nexis.birth_date_checks')
+      )
```

Comment:
for something like this (with 2 contain exactly calls), what if we just did a direct equality:
```suggestion
      expect(output).to eq(id: [i18n.t('doc_auth.errors.lexis_nexis.birth_date_checks')])
```

Reply:
Sure.

Reply:
To just add some documentation of the slack DM:

There is something weird here that happens with the `eq` method that causes it to not work as expected in this case. I am leaving this as is to maintain the exact matching nature.